lib/binding/kvdb_interface.c

```diff
@@ -1060,13 +1060,17 @@ hse_kvs_cursor_read(
     bool *                 eof)
 {
     merr_t err;
+    size_t vlen_dummy;
 
-    if (HSE_UNLIKELY(!cursor || !key || !klen || !val || !vlen || !eof || flags != 0))
+    if (HSE_UNLIKELY(!cursor || !key || !klen || !eof || flags != 0))
         return merr(EINVAL);
 
     err = ikvdb_kvs_cursor_read(cursor, flags, key, klen, val, vlen, eof);
     ev(err);
 
+    if (!vlen)
```

Comment:
Why is this necessary?

Reply:
I had added it for the perf counter update below, but I realize now that that is incorrect when we don't read the values. Fixed it.